CYD-Klipper/src/ui/serial/serial_console.cpp

```diff
@@ -0,0 +1,153 @@
+    {
+        static int index = 0;
+        int c;         // read character, -1 if none
+        int cnt = 100; // limit on amount of iterations in one go; we're supposed to be non-blocking!
+
+        while ((c = Serial.read()) != -1 && cnt > 0)
+        {
+            --cnt;
+
+            // backspace
+            if (c == 8)
+            {
+                if (index > 0)
+                {
+                    if(temporary_config.remote_echo) Serial.print("\x08 \x08"); // overwrite last character with space and move cursor 1 back.
+                    index--;
+                }
+                continue;
+            }
+
+            if(temporary_config.remote_echo) Serial.print((char)c); // echo
+
+            // Buffer overflow handling:
+            // start treating current buffer as invalid:
+            // - stop collecting more data
+            // - return false on delimeter, flushing everything collected,
+            // - restart collection from scratch after delimeter,
+            // - return control as normal.
+
+            if (index >= max_len - 1)
+            {
+                if (c == delimiter) // got delimeter: flush buffer quietly, restart collection.
+                {
+                    index = 0;
+                    return false;
+                }
+                else
+                    continue; // discard any data past the end of the buffer, keep reading
+            }
+
+            result[index++] = c;
+
+            // delimiter was found
+            if (c == delimiter)
+            {
+                result[index] = '\0'; // Null-terminate the string
+                index = 0;
+                return true; // Success: Delimiter found
+            }
+        }
+
+        return false; // still waiting for delimeter
+    }
+
+    // split input into words.
+    // assumes input ends with '\n', also ends parsing if found MAX_WORDS already.
+    int tokenize(String results[], char *input)
+    {
+        int index = 0;
+        int word_count = 0;
+        String word = "";
+        do
+        {
+            if (input[index] == '\t' || input[index] == ' ' || input[index] == '\n' || input[index] == '\r' || input[index] == '\0')
+            {
+                if (word.length() > 0)
+                {
+                    results[word_count] = word;
+                    ++word_count;
+                    if (word_count >= MAX_WORDS)
+                    {
+                        return word_count;
+                    }
+                    word = "";
+                }
+
+                if (input[index] == '\n' || input[index] == '\0')
+                {
+                    return word_count;
+                }
+                index++;
+            }
+            else
+            {
+                word += input[index];
+                index++;
+            }
+        } while (1);
+    }
+
+    void greet()
+    {
+        Serial.println("CYB-Klipper " REPO_VERSION);

```

Comment:
Cyb

Reply:
Sorry, fixed. I keep thinking it's a Cheap Yellow Board...